### PR TITLE
Logging to analytics

### PR DESCRIPTION
**Pull request by rock3r**
This PR adds Timber (WIP make sure we only use Timber to log stuff, will do in next PR) and refactors Analytics code to abstract away the specific implementations, so we can swap Google Analytics out and use Firebase Analytics instead (coming soon™). It also contains some cleanup and stuff.

### CODE REVIEW

File: app/src/main/java/com/connfa/model/http/hurl/HURLCookieStore.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
      */
     private Map<URI, List<HttpCookie>> mapCookies = new HashMap<URI, List<HttpCookie>>();
     /*
-     * The instance of the shared preferences
+     * The createInstance of the shared preferences
```
Comment (tiwiz) [inline app/src/main/java/com/connfa/model/http/hurl/HURLCookieStore.java:25]: Typo?

File: app/src/main/java/com/connfa/ui/activity/AboutDetailsActivity.java [hunk 2]
```diff
@@ -39,29 +26,14 @@ protected void onCreate(Bundle savedInstanceState) {
...
-        GoogleAnalytics.getInstance(this).reportActivityStop(this);
-    }
-
-    @Override
-    protected void onDestroy() {
-        super.onDestroy();
-        Model.instance().getUpdatesManager().unregisterUpdateListener(updateListener);
+        Analytics.from(this)
+                .trackEvent(
+                        getString(R.string.about_category), getString(R.string.action_open), id + " " + aboutTitle
```
Comment (tiwiz) [inline app/src/main/java/com/connfa/ui/activity/AboutDetailsActivity.java:35]: AS doesn't blame for the String composition? Shouldn't we use String.format?

File: app/src/main/java/com/connfa/model/http/hurl/HURLCookieStore.java (hunk 1 above)
Comment (rock3r) [reply app/src/main/java/com/connfa/model/http/hurl/HURLCookieStore.java:25]: Refactor went overboard /shrug

File: app/src/main/java/com/connfa/ui/activity/AboutDetailsActivity.java (hunk 2 above)
Comment (rock3r) [reply app/src/main/java/com/connfa/ui/activity/AboutDetailsActivity.java:35]: Can't be bothered, this will eventually go away anyway I think

**Review by tiwiz — APPROVED** [review]

**Pull request merged**